File: VenueBitApp/VenueBitApp/Core/Networking/Models/Venue.swift
import Foundation

struct Venue: Codable, Identifiable {
    let id: String
    let name: String
    let address: String?
    let city: String
    let state: String
    let zipCode: String?
    let capacity: Int?
    let type: VenueType?

    var cityState: String {
        "\(city), \(state)"
    }

    var fullAddress: String {
        if let address = address {
            return "\(address), \(city), \(state)"
        }
        return cityState
    }
}

enum VenueType: String, Codable {
    case stadium
    case arena
    case theater
}

File: api/_lib/types/venue.ts
export type VenueType = 'stadium' | 'arena' | 'theater';

export interface Venue {
  id: string;
  name: string;
  city: string;
  state: string;
  capacity: number;
  type: VenueType;
  address: string;
  imageUrl: string;
}
